pods-framework/pods
# Rework and fix field dependencies

## Description

Reworks field dependencies, adds extensive testing for dependencies. 

## Testing instructions

1. Add/Edit Field Modal > Advanced > Visibility > Restrict by Role / Capability: The dependent fields should show based on these checkboxes. 
2. For Relationship fields: test `pick_show_icon` which has a `depends-on-any` checking pick_format_single/pick_format_multi (see our Slack conversation)
3. Check other `depends-on` and `wildcard-on` and `exclude-on` fields. 

## Changelog text for these changes

- Bug Fix: Corrected logic for validating dependencies of dependencies when rendering or hiding a field. 
- Enhancement: Downgrades WordPress package dependencies to ones included in WP 5.5 for running Jest tests.
- Enhancement: Include package-lock.json in repo.
- Enhancement: Add tests for dependency validation to the FieldWrapper component.
- Enhancement: Add tests for dependency validation to the validation helper function.
- Enhancement: A few performance improvements to the "Edit Field" modals, using React's useMemo.

## PR checklist
